heron/downloaders/src/java/com/twitter/heron/downloader/FileDownloader.java

```diff
@@ -0,0 +1,26 @@
+//  Copyright 2017 Twitter. All rights reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+package com.twitter.heron.downloader;
+
+import java.net.URI;
+import java.net.URL;
+import java.nio.file.Path;
+
+public class FileDownloader implements Downloader {
```

Comment:
Why a new downloader is added? HttpDownloader is not enough in this case? The implementation seems to be the same.

If there is a new case, please add comment to explain. My concern is that in future people dont know which one to use or could mis-use them.

Reply:
So I added this because the heron_downloader couldn't handle URIs like: "file:///foo/bar".  This is the same implementation as the HttpDownloader, but I am not sure if we should point URIs with a prefix of "file" to the the http downloader or just have a new file downloader.  The reason to why we need to support copy of files by the heron_downloader is because there could be instances where the user is distributing JARs via some sort of network storage and we need to copy and extract the jars from the network storage location into the local heron directory structure

Reply:
Got it. thx. Please add a comment about the use case. An example could be good enough.

Reply:
sure thing